File: 0167_Two Sum II/TwoSumII.cs
public class Solution {
    public int[] TwoSum(int[] numbers, int target) {
        var ans = new int[2];
        
        int s = 0;
        int e = numbers.Length - 1;
        
        while(s<e){
            var sum = numbers[s] + numbers[e];
            if(sum == target){
                ans[0] = s+1;
                ans[1] = e+1;
                break;
            }else if(sum > target){
                e--;
            }else{
                s++;
            }
        }
        
        return ans;
    }
}

File: 0167_Two Sum II/TwoSumII.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int[] ans = new int[2];
        int n = numbers.length;
        int l = 0, r = n-1;
        
        while(l < r) {
            if(numbers[l] + numbers[r] == target) {
                ans[0] = l+1;
                ans[1] = r+1;
                break;
            }else if(numbers[l] + numbers[r] > target) {
                r--;
            }else {
                l++;
            }
        }
        
        return ans;
    }
}